### PR TITLE
[ECL] Full variance usage of vision data

### PR DESCRIPTION
**Pull request by kamilritz**
This adapts the interface to ECL EKF to support the use of all variance information in the visual_odometry message. More information can be found in the [ECL PR](https://github.com/PX4/ecl/pull/680).

This needs recently added function of Matrix. Therefore, it is also updating Matrix to newest version.

In addition, it is adding the missing transformation of the linear velocity covariances to local frame. This was undone before and therefore not completely correct. 

Testing:
Run estimation on vehicle, which does not have a true full covariance. But it still shows that the vision estimation is not broken by this PR. [Log](https://review.px4.io/plot_app?log=dec270d7-076b-46d6-b6e0-f1a4b2ed0cda)

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1288,11 +1287,25 @@ MavlinkReceiver::handle_message_odometry(mavlink_message_t *msg)
 		odometry.pitchspeed = odom.pitchspeed;
 		odometry.yawspeed = odom.yawspeed;
 
-		//TODO: Apply rotation matrix to transform from body-fixed NED to earth-fixed NED frame
+		/* the linear velocity's covariance needs to be transformed to the local frame FRD*/
```
Comment (mhkabir) [inline src/modules/mavlink/mavlink_receiver.cpp]: Seems like this was missing before (and therefore wrong/a bug). Can you make this it's own commit?

**Comment by mhkabir** [comment]
@kamilritz How did you test this?

File: msg/vehicle_odometry.msg [hunk 2]
```diff
@@ -17,7 +17,7 @@ uint8 COVARIANCE_MATRIX_YAWRATE_VARIANCE=20
 
 # Position and linear velocity frame of reference constants
 uint8 LOCAL_FRAME_NED=0         # NED earth-fixed frame
-uint8 LOCAL_FRAME_ENU=1         # ENU earth-fixed frame
+uint8 LOCAL_FRAME_FRD=1         # FRD earth-fixed frame, heading different to NED
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg]:
```suggestion
uint8 LOCAL_FRAME_FRD=1         # FRD earth-fixed frame, arbitrary heading reference point
```

File: src/modules/ekf2/ekf2_main.cpp [hunk 3]
```diff
@@ -1106,14 +1108,17 @@ void Ekf2::Run()
 				ev_data.vel(2) = _ev_odom.vz;
 
 				// velocity measurement error from ev_data or parameters
-				if (!_param_ekf2_ev_noise_md.get() && PX4_ISFINITE(_ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_VX_VARIANCE])) {
-					ev_data.velErr = fmaxf(_param_ekf2_evv_noise.get(),
-							       sqrtf(fmaxf(_ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_VX_VARIANCE],
-									   fmaxf(_ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_VY_VARIANCE],
-											   _ev_odom.pose_covariance[_ev_odom.COVARIANCE_MATRIX_VZ_VARIANCE]))));
+				float param_evv_noise_var = sq(_param_ekf2_evv_noise.get());
```
Comment (dagar) [inline src/modules/ekf2/ekf2_main.cpp]: Update the `EKF2_EVV_NOISE` parameter description to mention the new behaviour?

**Comment by kamilritz** [comment]
Updated the parameter description for `EKF2_EVV_NOISE`, `EKF2_EVP_NOISE` and `EKF2_EVA_NOISE`
